stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/feature/channel/info/ChannelInfoController.kt

```diff
@@ -0,0 +1,420 @@
+                ChannelInfoEvent.MuteChannelError(message = error.message),
+            )
+        }
+
+        requireCapability(
+            permission = { canMuteChannel },
+            onError = onError,
+        ) {
+            scope.launch {
+                channelClient.mute().await()
+                    .onError(onError)
+            }
+        }
+    }
+
+    /**
+     * Unmutes the channel for the current user.
+     */
+    public fun unmuteChannel() {
+        logger.d { "[unmuteChannel]" }
+
+        val onError: (Error) -> Unit = { error ->
+            logger.e { "[unmuteChannel] error: ${error.message}" }
+            _events.tryEmit(
+                ChannelInfoEvent.UnmuteChannelError(message = error.message),
+            )
+        }
+
+        requireCapability(
+            permission = { canMuteChannel },
+            onError = onError,
+        ) {
+            scope.launch {
+                channelClient.unmute().await()
+                    .onError(onError)
+            }
+        }
+    }
+
+    /**
+     * Hides the channel.
+     *
+     * @param clearHistory If true, the channel history will be cleared.
+     */
+    public fun hideChannel(clearHistory: Boolean) {
+        logger.d { "[hideChannel] clearHistory: $clearHistory" }
+        scope.launch {
+            channelClient.hide(clearHistory).await()
+                .onError { error ->
+                    logger.e { "[hideChannel] error: ${error.message}" }
+                    _events.tryEmit(
+                        ChannelInfoEvent.HideChannelError(message = error.message),
+                    )
+                }
+        }
+    }
+
+    /**
+     * Unhides the channel.
+     */
+    public fun unhideChannel() {
+        logger.d { "[unhideChannel]" }
+        scope.launch {
+            channelClient.show().await()
+                .onError { error ->
+                    logger.e { "[unhideChannel] error: ${error.message}" }
+                    _events.tryEmit(
+                        ChannelInfoEvent.UnhideChannelError(message = error.message),
+                    )
+                }
+        }
+    }
+
+    /**
+     * Leaves the channel with an optional quit message.
+     *
+     * @param quitMessage The system message to send when leaving the channel.
+     */
+    public fun leaveChannel(quitMessage: Message?) {
+        logger.d { "[leaveChannel] quitMessage: ${quitMessage?.text}" }
+
+        val onError: (Error) -> Unit = { error ->
+            logger.e { "[leaveChannel] error: ${error.message}" }
+            _events.tryEmit(
+                ChannelInfoEvent.LeaveChannelError(message = error.message),
+            )
+        }
+
+        requireCapability(
+            permission = { canLeaveChannel },
+            onError = onError,
+        ) {
+            scope.launch {
+                runCatching {
+                    requireNotNull(chatClient.getCurrentOrStoredUserId()) { "User not connected" }
```

Comment:
A user shouldn't be able to "leave a channel" if the user is not set (That is different that it is not connected)
I don't think this case needs to be handled, we can use the current user from the UserState and not adding extra code that won't be executer never

Reply:
All the operations can be called by the consumer at any time, so it's not guarantee that there will be always a connected user. That's why `getCurrentOrStoredUserId` is nullable.
I also used `getCurrentOrStoredUserId`  because it was simple to mock it in tests.

Reply:
Now, I understand what you mean. We should avoid using `getCurrentOrStoredUserId` because it might get the user from the cache even when it is not connected.
I will use `getCurrentUser` instead.
Thanks